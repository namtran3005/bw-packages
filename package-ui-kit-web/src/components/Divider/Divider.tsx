import * as React from 'react';

import styled from 'styled-components';
import { Divider as MuiDivider } from '@material-ui/core';
import { DividerProps as MuiDividerProps } from '@material-ui/core/Divider';

interface DividerProps extends MuiDividerProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const DividerComponent: React.FunctionComponent<DividerProps> = (props) => {
  return <MuiDivider {...props} />;
};

const Divider = styled(DividerComponent)`
  && {
    background-color: ${({ theme, ...props }) =>
      props.light
        ? theme.palette.get('backgroundLight')
        : theme.palette.getWithAlpha('primaryBlack', 0.25)};
  }
`;

export { Divider, DividerProps };
