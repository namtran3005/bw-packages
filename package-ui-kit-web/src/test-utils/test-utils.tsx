import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
