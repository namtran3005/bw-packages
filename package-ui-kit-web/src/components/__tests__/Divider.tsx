import * as React from 'react';
import { Divider } from '../Divider';
import { renderWithTestHarness } from '../../test-utils/test-utils';
const defaultProps = {};

describe('Divider', () => {
  it('should render divider', () => {
    const { baseElement } = renderWithTestHarness(
      <Divider {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
