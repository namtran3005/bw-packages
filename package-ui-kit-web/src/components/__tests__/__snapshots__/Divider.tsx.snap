// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render divider 1`] = `
.c0.c0 {
  background-color: rgba(44,35,46,0.25);
}

<body>
  <div>
    <hr
      class="MuiDivider-root c0"
    />
  </div>
</body>
`;
